### PR TITLE
Part of #12017 - create gfx_draw_sprite overload

### PR DESCRIPTION
**Pull request by TomasZilinek**
Part 1. I created the overload and replaced about a half of the calls since there are too many for one PR.

### CODE REVIEW

**Comment by tupaschoal** [comment]
Can you rebase first to solve the merge conflict, please?

**Comment by TomasZilinek** [comment]
> Can you rebase first to solve the merge conflict, please?

Is this it? I think I did it interactively here :D

File: src/openrct2/Intro.cpp [hunk 1]
```diff
@@ -183,10 +183,10 @@ void intro_draw(rct_drawpixelinfo* dpi)
...
 
             // Draw Infogrames logo
-            gfx_draw_sprite(dpi, SPR_INTRO_INFOGRAMES_00, { (screenWidth / 2) - 320 + 69 }, _introStateCounter + 69, 0);
-            gfx_draw_sprite(dpi, SPR_INTRO_INFOGRAMES_10, { (screenWidth / 2) - 320 + 319 }, _introStateCounter + 69, 0);
-            gfx_draw_sprite(dpi, SPR_INTRO_INFOGRAMES_01, { (screenWidth / 2) - 320 + 69 }, _introStateCounter + 319, 0);
-            gfx_draw_sprite(dpi, SPR_INTRO_INFOGRAMES_11, { (screenWidth / 2) - 320 + 319 }, _introStateCounter + 319, 0);
+            gfx_draw_sprite(dpi, SPR_INTRO_INFOGRAMES_00, { screenWidth / 2 - 320 + 69, _introStateCounter + 69 }, 0);
+            gfx_draw_sprite(dpi, SPR_INTRO_INFOGRAMES_10, { screenWidth / 2 - 320 + 319, _introStateCounter + 69 }, 0);
+            gfx_draw_sprite(dpi, SPR_INTRO_INFOGRAMES_01, { screenWidth / 2 - 320 + 69, _introStateCounter + 319 }, 0);
+            gfx_draw_sprite(dpi, SPR_INTRO_INFOGRAMES_11, { screenWidth / 2 - 320 + 319, _introStateCounter + 319 }, 0);
```
Comment (tupaschoal) [inline src/openrct2/Intro.cpp]: Restore the parantheses for the division

File: src/openrct2-ui/windows/About.cpp [hunk 2]
```diff
@@ -322,7 +322,7 @@ static void window_about_rct2_paint(rct_window* w, rct_drawpixelinfo* dpi)
     gfx_draw_string_centred(dpi, STR_LICENSED_TO_INFOGRAMES_INTERACTIVE_INC, screenCoords, COLOUR_BLACK, nullptr);
 
     // Images
-    gfx_draw_sprite(dpi, SPR_CREDITS_CHRIS_SAWYER_SMALL, w->windowPos.x + 92, yPage + 24, 0);
+    gfx_draw_sprite(dpi, SPR_CREDITS_CHRIS_SAWYER_SMALL, w->windowPos + ScreenCoordsXY{ 92, yPage + 24 }, 0);
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/About.cpp:325]: This seems incorrect, previous `y` was `yPage + 24`, now it is `w->windowPos.y + yPage + 24`

File: src/openrct2-ui/windows/EditorScenarioOptions.cpp [hunk 3]
```diff
@@ -440,20 +440,20 @@ static void window_editor_scenario_options_draw_tab_images(rct_window* w, rct_dr
...
         spriteIndex += (w->frame_no / 4) % 8;
 
-    gfx_draw_sprite(dpi, spriteIndex, w->windowPos.x + widget->left, w->windowPos.y + widget->top, 0);
+    gfx_draw_sprite(dpi, spriteIndex, w->windowPos.x + widget->left, w->windowPos.y, 0);
 
     // Tab 3
     widget = &w->widgets[WIDX_TAB_3];
     spriteIndex = SPR_TAB_PARK;
-    gfx_draw_sprite(dpi, spriteIndex, w->windowPos.x + widget->left, w->windowPos.y + widget->top, 0);
+    gfx_draw_sprite(dpi, spriteIndex, w->windowPos.x + widget->left, w->windowPos.y, 0);
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/EditorScenarioOptions.cpp:456]: These are not equivalent to what they were.

File: src/openrct2/Intro.cpp [hunk 4]
```diff
@@ -183,10 +183,10 @@ void intro_draw(rct_drawpixelinfo* dpi)
...
 
             // Draw Infogrames logo
-            gfx_draw_sprite(dpi, SPR_INTRO_INFOGRAMES_00, (screenWidth / 2) - 320 + 69, _introStateCounter + 69, 0);
-            gfx_draw_sprite(dpi, SPR_INTRO_INFOGRAMES_10, (screenWidth / 2) - 320 + 319, _introStateCounter + 69, 0);
-            gfx_draw_sprite(dpi, SPR_INTRO_INFOGRAMES_01, (screenWidth / 2) - 320 + 69, _introStateCounter + 319, 0);
-            gfx_draw_sprite(dpi, SPR_INTRO_INFOGRAMES_11, (screenWidth / 2) - 320 + 319, _introStateCounter + 319, 0);
+            gfx_draw_sprite(dpi, SPR_INTRO_INFOGRAMES_00, { screenWidth / 2 - 320 + 69, _introStateCounter + 69 }, 0);
+            gfx_draw_sprite(dpi, SPR_INTRO_INFOGRAMES_10, { screenWidth / 2 - 320 + 319, _introStateCounter + 69 }, 0);
+            gfx_draw_sprite(dpi, SPR_INTRO_INFOGRAMES_01, { screenWidth / 2 - 320 + 69, _introStateCounter + 319 }, 0);
+            gfx_draw_sprite(dpi, SPR_INTRO_INFOGRAMES_11, { screenWidth / 2 - 320 + 319, _introStateCounter + 319 }, 0);
```
Comment (tupaschoal) [inline src/openrct2/Intro.cpp:189]: Restore the parentheses

File: src/openrct2/Intro.cpp [hunk 5]
```diff
@@ -196,8 +196,8 @@ void intro_draw(rct_drawpixelinfo* dpi)
             gfx_clear(dpi, BACKROUND_COLOUR_DARK);
 
             // Draw Chris Sawyer logo
-            gfx_draw_sprite(dpi, SPR_INTRO_CHRIS_SAWYER_00, (screenWidth / 2) - 320 + 70, _introStateCounter, 0);
-            gfx_draw_sprite(dpi, SPR_INTRO_CHRIS_SAWYER_10, (screenWidth / 2) - 320 + 320, _introStateCounter, 0);
+            gfx_draw_sprite(dpi, SPR_INTRO_CHRIS_SAWYER_00, { screenWidth / 2 - 320 + 70, _introStateCounter }, 0);
+            gfx_draw_sprite(dpi, SPR_INTRO_CHRIS_SAWYER_10, { screenWidth / 2 - 320 + 320, _introStateCounter }, 0);
```
Comment (tupaschoal) [inline src/openrct2/Intro.cpp:200]: Restore the parentheses

File: src/openrct2/object/EntranceObject.cpp [hunk 6]
```diff
@@ -43,13 +43,12 @@ void EntranceObject::Unload()
...
-    int32_t x = width / 2;
-    int32_t y = height / 2;
+    auto screenCoords = ScreenCoordsXY{ width / 2, height / 2 };
 
     uint32_t imageId = _legacyType.image_id;
-    gfx_draw_sprite(dpi, imageId + 1, x - 32, y + 14, 0);
-    gfx_draw_sprite(dpi, imageId + 0, x + 0, y + 28, 0);
-    gfx_draw_sprite(dpi, imageId + 2, x + 32, y + 44, 0);
+    gfx_draw_sprite(dpi, imageId + 1, screenCoords + ScreenCoordsXY{ -32, 14 }, 0);
+    gfx_draw_sprite(dpi, imageId + 0, screenCoords + ScreenCoordsXY{ -0, 28 }, 0);
```
Comment (tupaschoal) [inline src/openrct2/object/EntranceObject.cpp:50]:
```suggestion
    gfx_draw_sprite(dpi, imageId + 0, screenCoords + ScreenCoordsXY{ 0, 28 }, 0);
```
No need for `-0` :D

**Comment by TomasZilinek** [comment]
Please help me :(

**Comment by frutiemax** [comment]
Your commit message is longer than 72 characters.

**Comment by TomasZilinek** [comment]
> Your commit message is longer than 72 characters.

It is not my commit message. Apparently, by rebasing, I inherited the commit by someone else which is incorrectly written.

**Comment by frutiemax** [comment]
You need to rebase and amend the incorrect commits by changing the message to a shorter one.

**Comment by tupaschoal** [comment]
@frutiemax is right, you're merging the develop onto your branch, you should be rebasing your branch against the develop instead. That would drop all the commits that are not yours or that are not from this branch and the "Lint Commit Message" would succeed

**Comment by tupaschoal** [comment]
I've fixed it for you @TomasZilinek . Please take some time to understand the difference between rebasing and merging, it will give you less headaches in the future: https://www.atlassian.com/git/tutorials/merging-vs-rebasing

What I did (after cloning your repo):
`git pull upstream develop` - Syncs your develop with ours
`git checkout issue12017fix` - Go to your branch
`git rebase -i develop` - Interactively rebases your commits on top of the updated develop branch. There, I deleted all the commits added by the merge action and kept just your first two. There were some conflicts that I had to solve until the rebase was finished
`git push -f origin` - Force pushes the rebased branch to GitHub

File: src/openrct2-ui/windows/GameBottomToolbar.cpp [hunk 7]
```diff
@@ -630,45 +632,46 @@ static void window_game_bottom_toolbar_draw_news_item(rct_drawpixelinfo* dpi, rc
             uint32_t image_id = image_id_base;
             image_id |= SPRITE_ID_PALETTE_COLOUR_2(peep->TshirtColour, peep->TrousersColour);
 
-            gfx_draw_sprite(&cliped_dpi, image_id, clip_x, clip_y, 0);
+            gfx_draw_sprite(&cliped_dpi, image_id, { clip_x, clip_y }, 0);
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/GameBottomToolbar.cpp:635]: Make `clip_x` and `clip_y` something like `clipCoords`

File: src/openrct2/paint/Paint.cpp [hunk 8]
```diff
@@ -522,17 +522,16 @@ static void paint_attached_ps(rct_drawpixelinfo* dpi, paint_struct* ps, uint32_t
...
         uint32_t imageId = paint_ps_colourify_image(attached_ps->image_id, ps->sprite_type, viewFlags);
         if (attached_ps->flags & PAINT_STRUCT_FLAG_IS_MASKED)
         {
-            gfx_draw_sprite_raw_masked(dpi, x, y, imageId, attached_ps->colour_image_id);
+            gfx_draw_sprite_raw_masked(dpi, screenCoords.x, screenCoords.y, imageId, attached_ps->colour_image_id);
         }
         else
         {
-            gfx_draw_sprite(dpi, imageId, x, y, ps->tertiary_colour);
+            gfx_draw_sprite(dpi, imageId, screenCoords.x, screenCoords.y, ps->tertiary_colour);
```
Comment (tupaschoal) [inline src/openrct2/paint/Paint.cpp:534]:
```suggestion
            gfx_draw_sprite(dpi, imageId, screenCoords, ps->tertiary_colour);
```

**Comment by TomasZilinek** [comment]
> I



> I've fixed it for you @TomasZilinek . Please take some time to understand the difference between rebasing and merging, it will give you less headaches in the future: https://www.atlassian.com/git/tutorials/merging-vs-rebasing
> 
> What I did (after cloning your repo):
> `git pull upstream develop` - Syncs your develop with ours
> `git checkout issue12017fix` - Go to your branch
> `git rebase -i develop` - Interactively rebases your commits on top of the updated develop branch. There, I deleted all the commits added by the merge action and kept just your first two. There were some conflicts that I had to solve until the rebase was finished
> `git push -f origin` - Force pushes the rebased branch to GitHub

Thank you. I made one mistake while rebasing the first time and then it just piled up. And you are right, this was one hell of a headache :D

**Pull request merged**